saleor/tests/e2e/checkout/discounts/promotions/test_promotion_applied_on_checkout_with_specific_subtotal.py

```diff
@@ -0,0 +1,143 @@
+    channel_id = shop_data["channel"]["id"]
+    channel_slug = shop_data["channel"]["slug"]
+    warehouse_id = shop_data["warehouse"]["id"]
+    shipping_method_id = shop_data["shipping_method"]["id"]
+
+    promotion_name = "Promotion"
+
+    promotion_data = create_promotion(e2e_staff_api_client, promotion_name)
+    promotion_id = promotion_data["id"]
+    discount_value = 2.99
+    order_predicate_subtotal_value = "21"
+    order_predicate = {
+        "discountedObjectPredicate": {
+            "subtotalPrice": {"eq": order_predicate_subtotal_value}
+        }
+    }
+
+    input = {
+        "promotion": promotion_id,
+        "channels": [channel_id],
+        "name": "test rule",
+        "orderPredicate": order_predicate,
+        "rewardType": "SUBTOTAL_DISCOUNT",
+        "rewardValue": discount_value,
+        "rewardValueType": "FIXED",
+    }
+
+    promotion_rule = create_promotion_rule(e2e_staff_api_client, input)
+    assert promotion_rule["orderPredicate"] == order_predicate
+    assert promotion_rule["channels"][0]["id"] == channel_id
+
+    return (
+        channel_id,
+        channel_slug,
+        shipping_method_id,
+        warehouse_id,
+        promotion_id,
+        discount_value,
+        order_predicate_subtotal_value,
+    )
+
+
+@pytest.mark.e2e
+def test_promotion_applied_on_checkout_with_specific_subtotal_CORE_2133(
+    e2e_staff_api_client,
+    e2e_not_logged_api_client,
+    shop_permissions,
+    permission_manage_product_types_and_attributes,
+    permission_manage_discounts,
+    permission_manage_orders,
+):
+    permissions = [
+        *shop_permissions,
+        permission_manage_product_types_and_attributes,
+        permission_manage_discounts,
+        permission_manage_orders,
+    ]
+    assign_permissions(e2e_staff_api_client, permissions)
+
+    (
+        channel_id,
+        channel_slug,
+        shipping_method_id,
+        warehouse_id,
+        promotion_id,
+        discount_value,
+        order_predicate_subtotal_value,
+    ) = prepare_promotion(e2e_staff_api_client)
+    (
+        _product_id,
+        product_variant_id,
+        product_variant_price,
+    ) = prepare_product(e2e_staff_api_client, warehouse_id, channel_id, variant_price=7)
+
+    # Step 1 - Create checkout with 5 product variants
+    lines = [{"variantId": product_variant_id, "quantity": 5}]
+    data = checkout_create(
+        e2e_not_logged_api_client,
+        lines,
+        channel_slug,
+        email="testEmail@example.com",
+        set_default_billing_address=True,
+        set_default_shipping_address=True,
+    )
+    checkout_id = data["id"]
+    total = data["totalPrice"]["gross"]["amount"]
+    assert data["billingAddress"] is not None
+    assert data["shippingAddress"] is not None
+    assert checkout_id is not None
+    assert data["lines"][0]["variant"]["id"] == product_variant_id
+    assert data["discount"]["amount"] == 0.00
+    assert total != order_predicate_subtotal_value
```

Comment:
Shouldn't we be checking the subtotal value here? I know that here, the total is equal to the subtotal because we don't have shipping added yet, but for better readability, we could check the value from the subtotal since we have it in the order predicate

Reply:
Correction applied @szczecha